ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/hardware-counters.j2

```diff
@@ -1,10 +1,13 @@
-{% if hardware_counters is defined and hardware_counters is not none %}
-{%     if hardware_counters.features is defined and hardware_counters.features is not none %}
+{# eos - Hardware counters #}
+{% if hardware_counters is arista.avd.defined %}
+{%     if hardware_counters.features is arista.avd.defined %}
 !
 {%         for feature in hardware_counters.features %}
 {%             for feature, direction in feature.items() %}
```

Comment:
Try to refactor this into something that looks like all the other templates. At least we should rename one of the `feature` variables, so we don't have the same name with different content.

Reply:
Variables renamed. I believe it makes sense to review this data model. Is the only place I found we use a list of dictionaries.

Reply:
Great. Can I ask you to submit an issue mentioning the datamodel and maybe a proposal for a better one? Thanks.